nucypher/blockchain/eth/sol/source/contracts/MiningAdjudicator.sol

```diff
@@ -144,7 +145,9 @@ contract MiningAdjudicator is Upgradeable {
         // Extract miner's address and check that is real miner
         // TODO: This will depend on the outcome of #962
         address miner = SignatureVerifier.recover(
-            SignatureVerifier.hash(_minerPublicKey, hashAlgorithm), _minerPublicKeySignature);
+            SignatureVerifier.hash(abi.encodePacked(precomp.lostBytes[4], miner_xcoord),
+                                   SignatureVerifier.HashAlgorithm.KECCAK256),
+            _minerPublicKeySignature);
         // Check that miner can be slashed
         (uint256 minerValue,,,,,) = escrow.minerInfo(miner);
```

Comment:
Is this the only way to know that the miner *can* be slashed?

Reply:
AFAIK, what this actually checks is that the miner is registered by `MinersEscrow`. @szotov is that right?

Reply:
And also that miner has something that we can take: miner can be slashed if he has stake, 
in #949 this line was changed to `getAlllTokens`